### PR TITLE
fix(morpho): remove unnecessary casts

### LINKED ISSUE #566: Unnecessary cast
There's also another unnecessary usage of `toUint128` at L372

_Originally posted by @Rubilmax in https://github.com/morpho-org/morpho-blue/pull/557#discussion_r1371725357_

### PR DESCRIPTION
**Pull request by Rubilmax**
- Fixes #566

### CODE REVIEW

**Review by QGarchery — APPROVED** [review]
Clean

**Review by MerlinEgalite — APPROVED** [review]

**Review by Jean-Grimal — APPROVED** [review]

File: src/Morpho.sol [hunk 1]
```diff
@@ -272,7 +273,8 @@ contract Morpho is IMorpho {
 
         position[id][onBehalf].borrowShares -= shares.toUint128();
         market[id].totalBorrowShares -= shares.toUint128();
-        market[id].totalBorrowAssets = UtilsLib.zeroFloorSub(market[id].totalBorrowAssets, assets).toUint128();
+        // Safe "unchecked" cast since the value is smaller than totalBorrowAssets.
+        market[id].totalBorrowAssets = uint128(UtilsLib.zeroFloorSub(market[id].totalBorrowAssets, assets));
```
Comment (MathisGD) [inline src/Morpho.sol:277]: I don't think that we should add think kind of optimization at this stage (even earlier in the development actually), similarly to #556. We don't know how much gas does it save right? (no veto though, it's correct indeed).

File: src/Morpho.sol [hunk 2]
```diff
@@ -369,12 +371,13 @@ contract Morpho is IMorpho {
 
         position[id][borrower].borrowShares -= repaidShares.toUint128();
         market[id].totalBorrowShares -= repaidShares.toUint128();
-        market[id].totalBorrowAssets = UtilsLib.zeroFloorSub(market[id].totalBorrowAssets, repaidAssets).toUint128();
+        // Safe "unchecked" cast since the value is smaller than totalBorrowAssets.
+        market[id].totalBorrowAssets = uint128(UtilsLib.zeroFloorSub(market[id].totalBorrowAssets, repaidAssets));
```
Comment (MathisGD) [inline src/Morpho.sol:375]: same

**Review by MathisGD — CHANGES_REQUESTED** [review]

File: src/Morpho.sol (hunk 1 above)
Comment (Rubilmax) [reply src/Morpho.sol:277]: I unfortunately agree that this PR has the exact same stakes than #556 so we shouldn't merge it... Too bad we realize late than we've complexified our lives

**Comment by MerlinEgalite** [comment]
Closing because we don't want to introduce small optimizations at this stage that we usually dismiss

**Pull request closed without merging**